arrow-libs/core/arrow-core/src/commonMain/kotlin/arrow/core/continuations/Raise.kt

```diff
@@ -286,6 +345,56 @@ public inline fun <R, B : Any> Raise<R>.ensureNotNull(value: B?, raise: () -> R)
 public inline fun <R, A, B> Raise<NonEmptyList<R>>.mapOrAccumulate(
   list: Iterable<A>,
   crossinline block: Raise<R>.(A) -> B
-): List<B> = list.mapOrAccumulate(Semigroup.nonEmptyList()) {
-  recover({ block(it) }, { raise(it.nel()) })
-}
+): List<B> =
+  list.mapOrAccumulate(Semigroup.nonEmptyList()) { elt -> mapErrorNel { block(elt) } }
+
+/**
+ * Accumulate the errors from running both [action1] and [action2].
+ */
+@EffectDSL
+public inline fun <R, A, B, C> Raise<NonEmptyList<R>>.zipOrAccumulate(
```

Comment:
For `Either` these are just called `zip`, should we also rename them to `zipOrAccumulate`? We removed regular `zip` since it's equivalent to `bind` for short-circuit behavior and we re-used the `zip` name for _accumulating_ behavior.

So we should be consistent between `Raise` and other data types.

Additionally, in `Either` is has signature `action1.zip(action2, block)` but it was requested on KotlinLang why it's not `Either.zip(action1, action2, block)` that would also align both these signatures further. It might even help back-porting them to 1.x.x easier, WDYT?

Reply:
I prefer too that version of `zip`, so I'm happy to make the change into a different PR, if you want.